app/apollo/resolvers/cluster.js

```diff
@@ -111,9 +110,9 @@ const clusterResolvers = {
           throw new NotFoundError(context.req.t('Could not find the cluster with Id {{clusterId}}.', {'clusterId':clusterId}), context);
         }
 
-        await validAuth(me, org_id, ACTIONS.READ, TYPES.CLUSTER, queryName, context, [clusterId, cluster.name]);
-
-        logger.info({req_id, user, org_id, clusterId, cluster }, `${queryName} found matching authorized cluster` );
+        logger.info({req_id, user, org_id, clusterId}, `${queryName} validating`);
+        await validAuth(me, org_id, ACTIONS.READ, TYPES.CLUSTER, queryName, context, [clusterId, cluster.registration.name || cluster.name.name]);
+        logger.info({req_id, user, org_id, clusterId}, `${queryName} validating - authorized`);
```

Comment:
- While the DB query can be before the `validAuth` call, the throwing of NotFoundError should be _after_, to avoid information leakage.
- The validAuth call will need to change it's `[ id, name ]` argument depending on whether the cluster was found in the DB or not.